lib/appraisal/matrix/rubygems_helper.rb

```diff
@@ -6,20 +6,43 @@
 module Appraisal
   module Matrix
     module RubygemsHelper
-      def versions_to_test(gem_name, minimum_version)
+      def versions_to_test(gem_name, minimum_version, maximum_version, step)
         # Generate a set to store the versions to test against 
         versions_to_test = Set.new
 
         # Load versions from rubygems api
         URI.parse("https://rubygems.org/api/v1/versions/#{gem_name}.json").open do |raw_version_data|
           JSON.parse(raw_version_data.read).each do |version_data|
             version = Gem::Version.new(version_data['number'])
-            versions_to_test << version.segments[0..1].join('.') if version >= minimum_version && !version.prerelease?
+            versions_to_test << version_segments(version, step).join('.') if include_version?(version, minimum_version, maximum_version)
           end
         end
 
         versions_to_test
       end
+
+      private
+
+      def version_segments(version, step)
+        case step
+        when :major
+          version.segments[0..0]
+        when :minor
+          version.segments[0..1]
+        when :patch
+          version.segments[0..2]
+        else
+          raise "Unsupported requested version step: #{step}"
+        end
+      end
+
+      def include_version?(version, minimum_version, maximum_version)
+        if maximum_version
+          !version.prerelease? && version >= minimum_version && version < maximum_version
```

Comment:
Shouldn't this be `<= maximum_version`?
```suggestion
          !version.prerelease? && version >= minimum_version && version <= maximum_version
```

Reply:
I think it's more effective for it to be `<` (this behavior is documented in the README as well). It wouldn't matter with major steps, but it'd be awkward with minor/patch steps to have to increment the appraisal file as new versions release or do something like `max: 7.9999` so that you don't bring in 8.0 but you always get the latest 7.x.

Reply:
`maximum_version` to me is inclusive. Perhaps a name change of `max` or we could refactor the code to allow `max` to be inclusive. Where a max of `8` means the highest version of `8.x`, 8.0` -> `8.0.x`

Reply:
I'm always up for a good rename. We could expect the key as `less_than` to be explicit? If you want the behavior described above, you can say `less_than: '8.1'` and there's no ambiguity about what versions are being included.

Reply:
I feel like we can just restructure the code to use `max` as inclusive, since we're already using `min`

Reply:
Also, an inclusive `max` allows us to not make up versions that don't exist yet